src/emulator/instruction.ts

```diff
@@ -213,91 +213,195 @@ function apply(opcode: number, instruction: Partial<Instruction>): void {
     };
 }
 
-function applySeriesR8_1(baseB: number, baseC: number, instruction: Partial<Instruction>): void {
-    [r8.b, r8.d, r8.h].forEach((reg, i) => apply(baseB + (i << 4), { ...instruction, par1: reg }));
-    [r8.c, r8.e, r8.l, r8.a].forEach((reg, i) => apply(baseC + (i << 4), { ...instruction, par1: reg }));
-}
+const instructions = new Array<Instruction>(0x100);
 
 for (let i = 0; i < 0x100; i++)
     instructions[i] = {
         opcode: i,
         operation: Operation.invalid,
-        addressingMode: AddressingMode.implicit,
-        par1: 0,
-        par2: 0,
         cycles: 0,
         len: 1,
     };
 
 apply(0, { operation: Operation.nop, cycles: 1, len: 1 });
-apply(0xc3, { operation: Operation.jp, addressingMode: AddressingMode.imm16, cycles: 4, len: 3 });
+apply(0xc3, { operation: Operation.jp, par1: { addressingMode: AddressingMode.imm16 }, cycles: 4, len: 3 });
```

Comment:
I would like to shorten  a few fields (see above) in order to keep those definitions in single lines --- I find this pretty hard to read otherwise. Note that prettier will not reformat automatically to single lines, you need to remove the first break --- it cost me a bit of digging to discover this last time it started splitting them 😏